### PR TITLE
Refactor predefined builders declaration to improve readability

### PR DESCRIPTION
**Pull request by zvirja**
Improved readability by extracting the verbose declaration to a helper methods. That makes it simpler to read those definitions and there are less chances to make a typo.

@moodmosaic Your review is welcome 😉

### CODE REVIEW

**Comment by zvirja** [comment]
@moodmosaic Fixed to make consistent 😉

File: Src/AutoFixture/Fixture.cs [hunk 1]
```diff
@@ -76,54 +76,47 @@ public Fixture(ISpecimenBuilder engine, MultipleRelay multiple)
...
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(SortedSet<>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(List<>))),
+
+                                MakeQueryBasedDictionaryBuilder(
```
Comment (moodmosaic) [inline Src/AutoFixture/Fixture.cs]: This is adding a level of indirection though, isn't it?

Comment (zvirja) [reply Src/AutoFixture/Fixture.cs]: Yep, it's adding. However that is not a big issue, at that heavily improves the readability. Instead parsing the whole graph to understand the supported types, you can simply look at the essential.

Previously we had the same for the "fixed builders":
https://github.com/AutoFixture/AutoFixture/blob/0e2075464a23f7cec5ddec92dfeaff5b5f32cf6e/Src/AutoFixture/Fixture.cs#L124-L126

I've extended that idea more and extracted repeated graph to a sub-method.

**Comment by zvirja** [comment]
@moodmosaic I've reworked the approach a bit (renamed a few methods), so please take a look. Also [I've replied](https://github.com/AutoFixture/AutoFixture/pull/952#discussion_r158291041) to your concern about indirection.

**Pull request merged**